pym/bob/input.py

```diff
@@ -2012,13 +2019,23 @@ def prepare(self, pathFormatter, inputEnv, sandboxEnabled, inputStates, inputSan
                     env.update(sandbox.getEnvironment())
                     if dep.provideGlobal: depEnv.update(sandbox.getEnvironment())
 
+        # if a dependency is a direct dependency with a empty use-section and also provided by some other package use the buildResult
+        for p in providedPackages:
```

Comment:
Now it's wrong. The `indirectPackages` list is the only correct one in this case. We pick up the results of provided packages only iff the package that provided them had `deps` in its use clause (which is the default). The whole hunk can be removed.